spec/support/page_titled.rb

```diff
@@ -0,0 +1,20 @@
+# Attempts to validate conformance to WCAG Success Criteria 2.4.2: Page Titled
+#
+# Visiting a page with the default app name title is considered a failure, and should be resolved by
+# providing a distinct description for the page using the `:title` content block.
+#
+# https://www.w3.org/WAI/WCAG21/Understanding/page-titled.html
+
+module PageTitled
+  def visit(*args)
+    super(*args)
+
+    raise "Visited page '#{current_path}' without distinct title" if title&.strip == APP_NAME
+  end
+end
+
+module Capybara
+  class Session
+    prepend PageTitled
+  end
+end
```

Comment:
This is definitely more thorough than our other accessibility specs, however, I don't like that we're monkeypatching like this so much

To add custom accessibility behavior, I added a custom matcher for labels (`label_required_fields`) and included it in every accessibility spec, WDYT of a similar approach?

ex: https://github.com/18F/identity-idp/blob/d16ae2f58e9f0a4f1611960a4a90da4d36b9dd47/spec/features/accessibility/idv_pages_spec.rb#L23-L24

Reply:
My worry is that it adds a lot of noise and is very easy to overlook when the developer's required to add it themselves, and consequently we'd just not have much assurance of this being implemented universally. There's a category of assertions I think could be good to run on every page load, regardless whether a developer writing tests knows they're being run. 

I agree that the monkey-patching isn't great though. I think the next best thing and more toward your suggestion could be one of:

- Changing each instance of `visit` to a helper `visit_and_validate_title` (or just `visit_and_validate`)
- Adapt `be_accessible` matcher to include this checking (or maybe it is already, or should be? 🤔 )

Or alternatively: Remove it altogether, now that the work is done 🤷  I tend to see this more of a thing of "How can we not need to do this again in the future".

Reply:
I'm a fan of @zachmargolis's approach

if our number of rules is growing, is it possible for us to define our own set of rules similar to `:section508` so we can tack it on to the list like `expect(page).to be_accessible.according_to :section508, :"best-practice", :login` ?

Reply:
Some related discussion in Slack on how we might customize this: https://gsa-tts.slack.com/archives/C0NGESUN5/p1605881043342700

Reply:
For the purposes of moving this forward, I think a custom matcher asserted in accessibility specs like originally suggested would probably fit best into our current approach. I don't think it would have caught all of the pages addressed here, and in general I don't know that we should need or want a dedicated set of accessibility specs (vs. absorbing them into the mainline feature specs), but it seems that there's a number of reasons we can't really do that for the time being.

Reply:
Yeah, after trying some stuff out in #4437, I think adding a new matcher under the label ones in the current specs may be the way to go. Then we can work on expanding the reach of our current accessibility specs and thinking through the longer term work we need to leverage the full feature test suite for accessibility checks.

Reply:
Updated to use new `be_uniquely_titled` matcher in accessibility specs in 0391a38.